diff --git a/atest/resources/testserver/testserver.py b/atest/resources/testserver/testserver.py
index 565c650..8e3d82b 100644
--- a/atest/resources/testserver/testserver.py
+++ b/atest/resources/testserver/testserver.py
@@ -3,6 +3,7 @@
 
 import os
 import sys
+from time import sleep
 
 from http.client import HTTPConnection
 from http.server import SimpleHTTPRequestHandler, HTTPServer
@@ -21,6 +22,36 @@ class StoppableHttpRequestHandler(SimpleHTTPRequestHandler):
     def do_POST(self):
         self.do_GET()
 
+    def do_GET(self):
+        """Response pages for Angular tests.
+        
+        Added by AngularJSLibrary
+        """
+        if self.path.endswith('/fastcall'):
+            self.send_response(200)
+            self.send_header('Content-type', 'text/html')
+            self.end_headers()
+            self.wfile.write('done')
+        elif self.path.endswith('/slowcall'):
+            sleep(2)
+            self.send_response(200)
+            self.send_header('Content-type', 'text/html')
+            self.end_headers()
+            self.wfile.write('finally done')
+        elif self.path.endswith('/fastTemplateUrl'):
+            self.send_response(200)
+            self.send_header('Content-type', 'text/html')
+            self.end_headers()
+            self.wfile.write(b'fast template contents')
+        elif self.path.endswith('/slowTemplateUrl'):
+            sleep(2)
+            self.send_response(200)
+            self.send_header('Content-type', 'text/html')
+            self.end_headers()
+            self.wfile.write(b'slow template contents')
+        else:
+            SimpleHTTPRequestHandler.do_GET(self)
+
 
 class ThreadingHttpServer(ThreadingMixIn, HTTPServer):
     pass
